src/base/utils/net.cpp

```diff
@@ -38,5 +38,27 @@ namespace Utils
         {
             return !QHostAddress(ip).isNull();
         }
+
+        bool isLoopbackAddress(const QHostAddress &addr)
+        {
+            return addr == QHostAddress::LocalHost
+                    || addr == QHostAddress::LocalHostIPv6
+                    || addr == QHostAddress(QLatin1String("::ffff:127.0.0.1"));
+        }
+
+        bool isLANAddress(const QHostAddress &addr)
+        {
+            // ipv4
+            if (addr.isInSubnet(QHostAddress(QLatin1String("0.0.0.0")), 0)) {
+                return addr.isInSubnet(QHostAddress(QLatin1String("10.0.0.0")), 8)
+                        || addr.isInSubnet(QHostAddress(QLatin1String("172.16.0.0")), 12)
+                        || addr.isInSubnet(QHostAddress(QLatin1String("192.168.0.0")), 16);
+            }
+            // ipv6 or ipv4-mapped ipv6
+            return addr.isInSubnet(QHostAddress(QLatin1String("::ffff:a00:0")), 104) // 10.0.0.0/8 mapped to ipv6
+                    || addr.isInSubnet(QHostAddress(QLatin1String("::ffff:ac10:0")), 108) // 172.16.0.0/12 mapped to ipv6
+                    || addr.isInSubnet(QHostAddress(QLatin1String("::ffff:c0a8:0")), 112) // 192.168.0.0/16 mapped to ipv6
+                    || addr.isInSubnet(QHostAddress(QLatin1String("fc00::")), 7);
+            }
```

Comment:
the `}` should have 4 less spaces right?
https://github.com/Piccirello/qBittorrent/blob/ea336e45804229c3cad5b4e82faf4a56e2cc8d21/src/base/utils/net.cpp#L62